### PR TITLE
Add canister_balance128

### LINKED ISSUE #373: IC API canister_balance128
canister_balance128: https://docs.rs/ic-cdk/latest/ic_cdk/api/fn.canister_balance128.html

### PR DESCRIPTION
**Pull request by dansteren**
Adds the canister_balance128 method.

BREAKING CHANGE: The `canisterBalance` method has changed to `canister_balance`. We will have a team discussion later on the pros/cons of camelCase vs snake_case. For now, the majority of the ledger and management canisters use snake_case and that's what the underlying rust implementation is, so we've normalized to that for now.

Resolves #373

### CODE REVIEW

**Review by lastmjs — CHANGES_REQUESTED** [review]
It looks good but let's wait until method_name is finished with tests and merged before we merge this.

**Comment by dansteren** [comment]
Today I learned about [`git rebase --onto`](https://stackoverflow.com/questions/21805925/accidentally-branched-off-of-the-wrong-branch-and-when-i-want-to-merge-into-the). That let me move the canister_balance128 commits off the message_name branch so that they instead are coming directlly from main. So this should now be good to merge in without having to wait for the message_name branch.

**Review by lastmjs — APPROVED** [review]

**Pull request merged**
